File: invision-backend/app/Enums/PaymentMethod.php
<?php

namespace App\Enums;

enum PaymentMethod: string
{
    case Cash = 'cash';
    case Check = 'check';
    case CreditCard = 'credit_card';
    case BankTransfer = 'bank_transfer';
    case Credit = 'credit';

    public function label(): string
    {
        return match ($this) {
            self::Cash => 'Cash',
            self::Check => 'Check',
            self::CreditCard => 'Credit Card',
            self::BankTransfer => 'Bank Transfer',
            self::Credit => 'Credit',
        };
    }
}

File: invision-mobile/lib/src/core/enums/payment_method.dart
enum PaymentMethod {
  cash,
  check,
  creditCard,
  bankTransfer,
  credit;

  String get label => switch (this) {
    cash => 'Cash',
    check => 'Check',
    creditCard => 'Credit Card',
    bankTransfer => 'Bank Transfer',
    credit => 'Credit',
  };

  static PaymentMethod fromString(String value) {
    return switch (value) {
      'cash' => PaymentMethod.cash,
      'check' => PaymentMethod.check,
      'credit_card' => PaymentMethod.creditCard,
      'bank_transfer' => PaymentMethod.bankTransfer,
      'credit' => PaymentMethod.credit,
      _ => PaymentMethod.cash,
    };
  }
}
